sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/LogicalRelation.scala

```diff
@@ -64,7 +64,8 @@ case class LogicalRelation(
   }
 
   override def simpleString(maxFields: Int): String = {
-    s"Relation[${truncatedString(output, ",", maxFields)}] $relation"
+    s"Relation ${catalogTable.map(_.identifier.unquotedString).getOrElse("")}" +
+      s"[${truncatedString(output, ",", maxFields)}] $relation"
```

Comment:
Can we fix DSv2 code path too? cc @MaxGekk, @imback82, @cloud-fan too

Reply:
It should already show the table name:
https://github.com/apache/spark/blob/e857e06452c2cf478beb31367f76d6950b660ebb/sql/catalyst/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2Relation.scala#L72

The question is should we show `identifier`.

Reply:
The problem is that it doesn't seem to show a table name: https://github.com/apache/spark/blob/e3058ba17cb4512537953eb4ded884e24ee93ba2/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/FileDataSourceV2.scala#L66-L71

Is it because `Table` API lets users to implement what name it returns, and Spark does not implement a catalog for its internal file sources? Seems like JDBC v2 supports its fully qualified name via `Identifier.toString`. If this is the case, I think it's fine.

Reply:
There are 2 "names" for v2 tables: the name to look up it and the name `Table` instance returns. It's possible that people use `tbl` to look up a v2 table and `Table.name` returns a file path. I think we should show both.

Reply:
Yes. It seems already show the table name. for example: `ParquetTable.name()` is table name.
https://github.com/apache/spark/blob/e857e06452c2cf478beb31367f76d6950b660ebb/sql/catalyst/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2Relation.scala#L67

https://github.com/apache/spark/blob/e857e06452c2cf478beb31367f76d6950b660ebb/sql/catalyst/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2Relation.scala#L123

https://github.com/apache/spark/blob/3d1dce75d96373130e27b3809c73d3796b5b77be/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/parquet/ParquetDataSourceV2.scala#L34-L36

Reply:
Okay, I am fine.